Source/WebKit/UIProcess/API/Cocoa/_WKTextExtraction.swift

```diff
@@ -559,10 +567,29 @@ extension WKTextExtractionTextItem {
         var parts = super.textRepresentationParts
 
         if !content.isEmpty {
-            parts.insert("text='\(content.escaped)'", at: 0)
-        }
+            let trimmedContent = content.trimmingCharacters(in: CharacterSet.newlines)
+            let leadingNewlineCount = content.prefix { $0.isNewline }.count
+            var adjustedSelectedRange = selectedRange
```

Comment:
In Swift, `Range` and `String.Index` should always be preferred over `NSRange`. (In this case, you could do

```
guard var adjustedSelectedRange = Range(selectedRange, in: content) else {
// handle case where Range is nil because it is invalid
}

// use adjustedSelectedRange and string indices for calculations
```